src/app/trade/page.tsx

```
@@ -35,117 +40,135 @@ import {
-  const { isLoading: isApprovalConfirming, isSuccess: isApprovalConfirmed } = useWaitForTransactionReceipt({ 
-    hash: approvalHash 
-  });
-
-
+  const { isLoading: isApprovalConfirming, isSuccess: isApprovalConfirmed } =
+    useWaitForTransactionReceipt({
+      hash: approvalHash,
+    });
 
   // Read contract data
   const { data: baseCoinAddress } = useReadContract({
```

Biome (2.4.4)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

(lint/correctness/useHookAtTopLevel)